HabitRPG/Views/HRPGCurrencyCountView.swift

```diff
@@ -85,14 +98,41 @@ class HRPGCurrencyCountView: UIView {
                                               attribute: NSLayoutAttribute.centerY,
                                               multiplier: 1,
                                               constant: 0))
-        currencyImageView.addConstraint(
-            NSLayoutConstraint.init(item: currencyImageView,
-                                    attribute: NSLayoutAttribute.width,
-                                    relatedBy: NSLayoutRelation.equal,
-                                    toItem: nil,
-                                    attribute: NSLayoutAttribute.notAnAttribute,
-                                    multiplier: 1,
-                                    constant: 18))
+        
+        widthConstraint = NSLayoutConstraint.init(item: currencyImageView,
+                                                  attribute: NSLayoutAttribute.width,
+                                                  relatedBy: NSLayoutRelation.equal,
+                                                  toItem: nil,
+                                                  attribute: NSLayoutAttribute.notAnAttribute,
+                                                  multiplier: 1,
+                                                  constant: viewSize == .normal ? 18 : 24)
+        heightConstraint = NSLayoutConstraint.init(item: currencyImageView,
+                                                   attribute: NSLayoutAttribute.height,
+                                                   relatedBy: NSLayoutRelation.equal,
+                                                   toItem: nil,
+                                                   attribute: NSLayoutAttribute.notAnAttribute,
+                                                   multiplier: 1,
+                                                   constant: viewSize == .normal ? 16 : 20)
+        if let width = widthConstraint {
+            currencyImageView.addConstraint(width)
+        }
+        if let height = heightConstraint {
+            currencyImageView.addConstraint(height)
+        }
+        
+        setNeedsUpdateConstraints()
+        updateConstraints()
+        setNeedsLayout()
+        layoutIfNeeded()
+    }
+    
+    private func updateSize() {
+        widthConstraint?.constant = (viewSize == .normal ? 18 : 24)
+        heightConstraint?.constant = (viewSize == .normal ? 16 : 20)
+        
+        countLabel.font = viewSize == .normal ? UIFont.preferredFont(forTextStyle: .footnote) : countLabel.font.withSize(17)
+        
+        currencyImageView.image = (viewSize == .normal ? #imageLiteral(resourceName: "Gem") : #imageLiteral(resourceName: "big_gem"))
```

Comment:
doesn't this set the icon to the gem every time the size is explicitly set?
Also, for the gem there is a paintcode version in the HabiticaIcons class, that should be used instead. (the big gem should also be there, if not let me know)

Reply:
Good catch! Will update on both these counts.